client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,136 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import omitBy from 'lodash/omitBy';
+import pick from 'lodash/pick';
+import includes from 'lodash/includes';
+import QueryString from 'querystring';
+
+/**
+ * Internal dependencies
+ */
+import shortcodeUtils from 'lib/shortcode';
+import { getVideo } from 'state/videos/selectors';
+import QueryVideo from 'components/data/query-video';
+
+class WpVideoView extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constrainVideoDimensions( shortcodeWidthAttribute, shortcodeHeightAttribute ) {
+		const defaultWidth = 640;
+		const defaultAspectRatio = 16 / 9;
+		let { width = defaultWidth, height = defaultWidth / defaultAspectRatio } = this.props.video || {};
+		const aspectRatio = width / height;
```

Comment:
This is the real aspect ratio of the video which might be different from the default (i.e. 4:3, and not 16:9). If `this.props.video` is undefined then `aspectRatio` and `defaultAspectRatio` are the same, but in general this is not true.